feat(serverless): update concepts (#4150)
scaleway/docs-content: serverless/jobs/concepts.mdx

  - serverless
---

## Container Registry

> Container Registry is the place where your images are stored before being deployed, we recommend using Scaleway Container Registry for optimal integration. [Migration guide](/serverless/containers/api-cli/migrate-external-image-to-scaleway-registry/).

Reviewer: ```suggestion
Container Registry is the place where your images are stored before being deployed. We recommend using Scaleway Container Registry for optimal integration. See the [migration guide](/serverless/containers/api-cli/migrate-external-image-to-scaleway-registry/) for full details.
```

A job run is the execution of a job definition. It can be in a running, succeeded, canceled, or failed status. Each job run has a unique identifier and can be individually monitored using [Cockpit](/observability/cockpit/quickstart/).

## Logging

> Serverless offers a built-in logging system based on cockpit to follow the activity of your ressources: [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).

Reviewer 2: ```suggestion
Serverless offers a built-in logging system based on cockpit to follow the activity of your resources: [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).
```

> Serverless offers a built-in logging system based on cockpit to follow the activity of your ressources: [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).

Reviewer: ```suggestion
Serverless offers a built-in logging system based on Scaleway Cockpit to track the activity of your resources: see [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).
```

## Maximum duration

The maximum duration option allows you to define the maximum execution time before your job is automatically killed.

## Metrics

> Performances of your Serverless ressources are natively available: [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).

Reviewer 2: ```suggestion
Performances of your Serverless resources are natively available: [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).
```

> Performances of your Serverless ressources are natively available: [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).

Reviewer: ```suggestion
Performance metrics for your Serverless resources are natively available: see [monitoring Serverless Jobs](/serverless/jobs/how-to/monitor-job/).
```

## Schedule (cron)

A schedule (cron) is a mechanism used to automatically start a Serverless Job at a specific time on a recurring schedule. It works similarly to a traditional Linux cron job, using the `* * * * *` format. Refer to our [cron schedules reference](/serverless/jobs/reference-content/cron-schedules/) for more information.

## Secrets reference

A secret reference is a mechanism that allows you to use a secret stored in [Secret Manager](/identity-and-access-management/secret-manager/) within Serverless Jobs. It allows you to securely reference sensitive data, such as API secret keys, passwords, tokens, or certificates.

## Startup command

This optional field allows you to specify a custom command executed upon starting your job if your container image does not have one already, or if you use a public container image.

## Status

> A Serverless Job Run can have the following statuses:

Reviewer 2: ```suggestion
A Serverless Job run can have the following statuses:
```

> * **Succeeded**: your Serverless Job Run finished in a good state.

Reviewer: ```suggestion
* **Succeeded**: your Serverless Job Run finished in a successful state.
```

> * **Queued**: your Serverless Job Run is waiting for ressources to run.

Reviewer 2: ```suggestion
* **Queued**: your Serverless Job Run is waiting for resources to run.
```

> * **Queued**: your Serverless Job Run is waiting for ressources to run.

Reviewer: ```suggestion
* **Queued**: your Serverless Job Run is waiting for resources to run.
```

> * **Error**: your Serverless Job Run finished with an error or timeout. [Check our troubleshooting documentation](/serverless/jobs/troubleshooting/job-in-error-state/) to solve the issue.

Reviewer 2: ```suggestion
* **Error**: your Serverless job run finished with an error or timeout. [Check our troubleshooting documentation](/serverless/jobs/troubleshooting/job-in-error-state/) to solve the issue.
```

> * **Canceled**: your Serverless Job Run has been canceled by the user.

Reviewer 2: ```suggestion
* **Canceled**: your Serverless job run has been canceled by the user.
```

## Stateless

Refers to a system or application that does not maintain any persistent state between executions. In a stateless environment, each request or operation is independent, and no information is retained from previous interactions.

This means that each request is treated as a new and isolated event, and there is no need for the system to remember previous states or data once a task is completed. Statelessness is commonly used in serverless architectures where each function execution is independent of others.

> To store informations you can use [Scaleway Object Storage](/storage/object/), [Scaleway Managed Databases](/managed-databases/postgresql-and-mysql/), [Scaleway Serverless Databases](/serverless/sql-databases/).

Reviewer: ```suggestion
To store data you can use [Scaleway Object Storage](/storage/object/), [Scaleway Managed Databases](/managed-databases/postgresql-and-mysql/), and [Scaleway Serverless Databases](/serverless/sql-databases/).
```
